static/app/components/forms/formPanel.tsx

```diff
@@ -57,11 +69,86 @@ function FormPanel({
   renderHeader,
   collapsible,
   initiallyCollapsed = false,
+  nested = false,
   ...otherProps
 }: FormPanelProps) {
   const [collapsed, setCollapse] = useState(initiallyCollapsed);
   const handleCollapseToggle = useCallback(() => setCollapse(current => !current), []);
 
+  const panelBody = (
+    <Fragment>
+      {typeof renderHeader === 'function' && renderHeader({title, fields})}
+
+      {fields.map(field => {
+        if (typeof field === 'function') {
+          return field();
+        }
+
+        const {defaultValue: _, ...fieldWithoutDefaultValue} = field;
+        const fieldConfig =
+          field.type === 'boolean' || field.type === 'bool'
+            ? field
+            : fieldWithoutDefaultValue;
+
+        // Allow the form panel disabled prop to override the fields
+        // disabled prop, with fallback to the fields disabled state.
+        if (disabled === true) {
+          fieldWithoutDefaultValue.disabled = true;
+          fieldWithoutDefaultValue.disabledReason = undefined;
+        }
+
+        return (
+          <FieldFromConfig
+            access={access}
+            disabled={disabled}
+            key={field.name}
+            {...otherProps}
+            {...additionalFieldProps}
+            field={fieldConfig}
+            highlighted={otherProps.highlighted === `#${field.name}`}
+          />
+        );
+      })}
+      {typeof renderFooter === 'function' && renderFooter({title, fields})}
+    </Fragment>
+  );
+
+  if (nested) {
+    return (
+      <Container padding="lg xl" data-test-id="collapsible-section-container">
+        <Flex padding="sm 0" data-test-id="header">
+          {title && (
+            <Button
+              priority="link"
```

Comment:
should we even be using our custom button here? maybe a styled `button` would be more appropriate.